lib/puppet/daemon.rb

```diff
@@ -168,7 +168,7 @@ def remove_pidfile
   def run_event_loop
     agent_run = Puppet::Scheduler.create_job(Puppet[:runinterval], Puppet[:splay], Puppet[:splaylimit]) do
       # Splay for the daemon is handled in the scheduler
-      agent.run(:splay => false)
+      agent.run(:splay => false, :waitforlock => Puppet[:waitforlock])
```

Comment:
Do we need to pass `:waitforlock` around? From what I see we always use it from `Puppet[:<setting>]`, so it should be available everywhere.